index.js

```diff
@@ -16,18 +22,60 @@ async function run() {
     const logConfigurationOptions = core.getInput("log-configuration-options", { required: false });
     const dockerLabels = core.getInput('docker-labels', { required: false });
     const command = core.getInput('command', { required: false });
+    
+     //New inputs 
+     const taskDefinitionArn = core.getInput('task-definition-arn', { required: false });
+     const taskDefinitionFamily = core.getInput('task-definition-family', { required: false });
+     const taskDefinitionRevision = Number(core.getInput('task-definition-revision', { required: false}));
+    
+
+    if (taskDefinitionFile && taskDefinitionArn) { // checks if provided both task definition and task definition arn
+      core.warning("Both task definition file and task definition arn are provided: task definition file will be option used.");
+    }
+
+    if(taskDefinitionArn){
+      core.debug("The specific amazon resource name is used to fetch your task definition");
+    }
+
+    if(taskDefinitionFamily && taskDefinitionRevision){
+      core.warning("Both task definition family and task definition revision are provided: the most up to date version will be used to fetch task definition");
+    }
 
-    // Parse the task definition
-    const taskDefPath = path.isAbsolute(taskDefinitionFile) ?
-      taskDefinitionFile :
-      path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile);
-    if (!fs.existsSync(taskDefPath)) {
-      throw new Error(`Task definition file does not exist: ${taskDefinitionFile}`);
+    if(taskDefinitionFamily && !taskDefinitionRevision){
+      throw new Error("Provide task definition revision if task definition family will be used to fetch task definition - vice versa ");
     }
+
+
+    // task definition arn, revison and family become part of the selection 
+    let describeTaskDefResponse;
```

Comment:
There needs to be check here whether or not `describeTaskDefinition` should be called, based on customer input. Additionally `describeTaskDefResponse` is not being used anywhere.